examples/simple_editor.rs

```diff
@@ -0,0 +1,110 @@
+#![cfg_attr(not(feature = "gtk_3_10"), allow(unused_variables, unused_mut))]
+
+extern crate time;
+extern crate gtk;
+extern crate gdk;
+extern crate cairo;
+extern crate lspace;
+
+use std::string::String;
+use std::rc::Rc;
+
+use gtk::traits::*;
+use gtk::signal::Inhibit;
+
+// for a list of key names see: http://gtk-rs.org/docs/gdk/enums/key/index.html
+use gdk::enums::key;
+
+use lspace::geom::colour::Colour;
+use lspace::elements::element::{ElementRef, elem_as_ref};
+use lspace::elements::container_sequence::TContainerSequenceElement;
+use lspace::elements::{text_element, row, column};
+use lspace::input::keyboard::{KeyEventType, KeyEvent, TKeyboardInteractor};
+use lspace::lspace_widget::LSpaceWidget;
+use lspace::lspace_area::LSpaceArea;
+
+
+struct LineEditor {
+    text_elem: ElementRef
+}
+
+impl LineEditor {
+    pub fn new(elem: &ElementRef) -> Rc<LineEditor> {
+        Rc::new(LineEditor{text_elem: elem.clone()})
+    }
+}
+
+impl TKeyboardInteractor for LineEditor {
+    fn on_key_event(&self, event: &KeyEvent) {
+        if event.event_type() == KeyEventType::Press {
+            let mut text = self.text_elem.as_text_element().unwrap().get_text().clone();
+            println!("Key val = {}", event.key_val());
+            if event.key_val() == key::BackSpace as u32 {
```

Comment:
This feels like it might better be a `match` statement (which will save continually calling `event.key_val()`).